provider/google.go

```diff
@@ -200,44 +186,45 @@ func (p *googleProvider) Records(zone string) (endpoints []*endpoint.Endpoint, _
 		return nil
 	}
 
-	err := p.resourceRecordSetsClient.List(p.project, zone).Pages(context.TODO(), f)
-	if err != nil {
-		return nil, err
+	for _, z := range zones {
+		if err := p.resourceRecordSetsClient.List(p.project, z.Name).Pages(context.TODO(), f); err != nil {
+			return nil, err
+		}
 	}
 
 	return endpoints, nil
 }
 
 // CreateRecords creates a given set of DNS records in the given hosted zone.
-func (p *googleProvider) CreateRecords(zone string, endpoints []*endpoint.Endpoint) error {
+func (p *googleProvider) CreateRecords(endpoints []*endpoint.Endpoint) error {
 	change := &dns.Change{}
 
 	change.Additions = append(change.Additions, newRecords(endpoints)...)
 
-	return p.submitChange(zone, change)
+	return p.submitChange(change)
 }
 
 // UpdateRecords updates a given set of old records to a new set of records in a given hosted zone.
-func (p *googleProvider) UpdateRecords(zone string, records, oldRecords []*endpoint.Endpoint) error {
+func (p *googleProvider) UpdateRecords(records, oldRecords []*endpoint.Endpoint) error {
 	change := &dns.Change{}
 
 	change.Additions = append(change.Additions, newRecords(records)...)
 	change.Deletions = append(change.Deletions, newRecords(oldRecords)...)
 
-	return p.submitChange(zone, change)
+	return p.submitChange(change)
 }
 
 // DeleteRecords deletes a given set of DNS records in a given zone.
-func (p *googleProvider) DeleteRecords(zone string, endpoints []*endpoint.Endpoint) error {
+func (p *googleProvider) DeleteRecords(endpoints []*endpoint.Endpoint) error {
 	change := &dns.Change{}
 
 	change.Deletions = append(change.Deletions, newRecords(endpoints)...)
 
-	return p.submitChange(zone, change)
+	return p.submitChange(change)
 }
 
 // ApplyChanges applies a given set of changes in a given zone.
-func (p *googleProvider) ApplyChanges(zone string, changes *plan.Changes) error {
+func (p *googleProvider) ApplyChanges(_ string, changes *plan.Changes) error {
```

Comment:
this argument is planned to be dropped once AWS will support multi hosted zone right ?